lib/daemon/index.js

```diff
@@ -1,45 +1,56 @@
 "use strict";
 
-const tap = require("lodash/fp").tap;
 const cpus = require("os").cpus;
 const path = require("path");
 const workerpool = require("workerpool");
 
 const actions = require("../actions");
 const Cache = require("../utils/cache");
-const hash = require("../utils/hash");
 
 class InspectpackDaemon {
   /**
    * Start the daemon with an asynchronously initialized cache.
    *
    * @param {Object} opts                  Object options
-   * @param {Object} opts.cacheFilename    The filename of the daemon cache
+   * @param {Object} opts.cache            A cache instance
    * @returns {InspectpackDaemon} The daemon with cache
    */
   static create(opts) {
-    opts = opts || {};
-    const cache = Cache.create({ filename: opts.cacheFilename });
-    return new InspectpackDaemon(cache, opts);
+    return new InspectpackDaemon(opts);
   }
 
   /**
    * Start the daemon with either no cache or an existing cache.
    *
-   * @param {Cache} cache                  An existing cache instance
-   * @param {Object} opts                  Object options
-   * @param {Object} opts.cacheFilename    The filename of the daemon cache
+   * @param {Object} opts                 Object options
+   * @param {Object} opts.cache           A cache instance
+   * @param {Object} opts.cacheFilename   A filename to create a cache if not provided.
+   * @returns {InspectpackDaemon}         The daemon with cache
    */
-  constructor(cache, opts) {
-    this._cache = cache || null;
-    this._cacheFilename = opts.cacheFilename || null;
+  constructor(opts) {
+    opts = opts || {};
+    this._cache = opts.cache ||
+      (opts.cacheFilename ? Cache.create({ filename: opts.cacheFilename }) : null);
+    if (!this._cache) {
+      throw new Error("Cache is required");
+    }
+
     this._pool = workerpool.pool(path.resolve(__dirname, "worker.js"), {
       minWorkers: cpus().length,
       maxWorkers: cpus().length,
-      forkArgs: [`cacheFilename=${this._cacheFilename}`]
+      forkArgs: this._cache ? [JSON.stringify({ cache: this._cache.serialize() })] : []
     });
   }
 
+  exec(action, optsArr) {
+    return this._pool
+      .exec(action, optsArr)
+      .catch((err) => {
+        console.error("Worker error", err); // eslint-disable-line no-console
+        this.terminate();
```

Comment:
@tptee -- I put this in because we're missing error capture for things that come in the callback. 

My one question is -- should I `terminate()` here? Or leave that upstream to the Daemon caller?

Reply:
Upstream should `terminate()` since not all of these errors will be fatal/require tearing down the pool. Although it'd be nice to find a way to distinguish that...

Reply:
OK. I'll remove.